### DISCUSSION TITLE
Best strategy to create aliases or redirects?

### DISCUSSION THREAD

**Discussion by shah** [lumeland/lume#127] [Q&A]
In my Hugo pages I have the ability to have a single markdown page be accessible through one or more "aliases" (basically redirects). Almost like "symlinks" so we don't duplicate the content in two URLs but one URL does an HTTP 302 redirect to another. What's Lume strategy for doing something similar.

For example, imagine I have `/my-site/users/shahid-shah` which I want to be a "friendly" alias or redirect to `/my-site/users/1` which is the actual content page. How would I accomplish this?

**Comment by oscarotero** [comment]
This is a server configuration, not related with the static site.
Most static sites hosts have `_redirect` and `_headers` to configure redirections and headers sent by the server. For example netlify: https://docs.netlify.com/routing/redirects/ or cloudflare https://developers.cloudflare.com/pages/platform/redirects

So the strategy for lume is simply create a _redirect file:

```js
export const url = "_redirect";

export default function () {
  const lines = [];
  for (const [from, to] of redirects) {
    lines.push(`${from} ${to} 301`);
  }
  return lines.join("\n");
}
```

**Reply by shah** [reply]
Got it thanks @oscarotero - but we like to be server-independent so it works in sandbox as well as in servers. In Hugo we're fully portable (not server-dependent) with directs because it generates HTML redirect files for aliases. I might create a new plugin to do the same for Lume if it's not already built in.

**Reply by oscarotero** [reply]
Do you mean creating html pages with a `http-equiv="refresh" content="0; url={{ newUrl }}"`?
It should work but looks a bit hacky to me.

**Reply by shah** [reply]
Yes that's what I mean -- but it's not really "hacky" as Google recognizes it as equivalent to 302 redirects for SEO purposes. Using anything server-dependent means we're stuck with Netlify, Vercel, etc. and cannot make it testable on any vanilla server (like sandbox) or move to Deno Deploy when desired. We could create a plugin that generates aliases and redirects across all server types as well as generate `http-equiv` by passing in arguments. 

But, I strongly recommend that redirects and aliases to pages be handled as a first-class _concept_ in Lume because anything more than a few pages on a basic site that has to live in a SEO-friendly environment needs that capability. As long as Lume as an opinionated way of handling redirects and 302s for SEO (whether it's a plugin or not is not important, the core needs to know the same page can live in multiple places and SEO scores won't get punished for duplication or 404s).
